### ISSUE TITLE
Not possible to check updates behind corporate ZScaler proxy

### ISSUE THREAD

**Issue by fordev4dev** [Zettlr/Zettlr#5271]
### Description

Trying to check updates reports the following error: "UNABLE_TO_GET_ISSUER_CERT_LOCALLY: Could not check for updates. RequestError: unable to get local issuer certificate". It is probably caused by my corporate ZScaler proxy which is working as MITM - replacing certificates for all sites by certificates issued by ZScaler itself.

### Reproducing

Probably possible only behind Zscaler or other corporate security suite working as described above

### Zettlr version

3.2.0 Windows

### Installation Method

From the Website or GitHub

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit (includes Apple Silicon)

### Operating System Version

Windows 10 Enterprise 22H2

**Comment by W4JEW** [comment]
This issue applies to any case where SSL interception/decryption is employed — not just Zscaler. I am also experiencing this; however, it's with Cloudflare's Zero Trust solution.

Ideally, Zettlr should use the operating system certificate store instead of maintaining its own. I recognize that might be a significant effort. At the very least, please provide a way to add additional Root CA certificates.

**Comment by nathanlesage** [comment]
Zettlr does not maintain any certificate store, or do any config at all. Instead, I merely use the corresponding API calls in Electron for any network calls. There could be issues with the intermediate layer, `got`, however I can't really test this out as I have no such configuration here and can't test it properly.

**Comment by nathanlesage** [comment]
So, I just did some research, but the internet is relatively scarce with information in this regard. What I could find are some very vague issues and problem descriptions. What I see are three potential ways forward:

1. I need a way to reliably test these issues to try out various ways of making network connections to see which ones honor the system stack, and which ones don't.
2. You can also tell me if there are other Electron-based apps that are Open Source and whose update mechanism works on your systems, then I can try to work myself backward from there.
3. Someone of you lends me a hand, I'm happy to merge PRs that address these issues.

**Comment by leebehrens** [comment]
@nathanlesage @fordev4dev After a couple days' poking I figured out one solution that is working for me.

The following assumes Zettlr is installed at C:\Users\<username>\AppData\Local\Programs\Zettlr.

1. Create batch file `C:\Users\<username>\AppData\Local\Programs\launch_zettlr.bat`:
   ```
   @echo off
   set NODE_TLS_REJECT_UNAUTHORIZED=0
   start "" "C:\Users\<username>\AppData\Local\Programs\Zettlr\Zettlr.exe"
   ```
2. Find the Zettlr shortcut in the Start menu. Right-click it and choose "Open file location" from the context menu.
3. The Windows File Explorer opens, with the Zettlr shortcut file selected. Right-click the shortcut and choose "Properties" from the context menu.
4. In the Properties window, change the Target to `C:\Users\<username>\AppData\Local\Programs\launch_zettlr.bat`
5. Leave Start in as is (`C:\Users\<username>\AppData\Local\Programs\Zettlr`).
6. Optional: Set Run to Minimized. (This will prevent a command window from briefly appearing while Zettlr launches.)
7. Click OK. (If an error pops up, correct the Target path to the .bat file in step 4.)
8. Launch Zettlr from the Start menu.
9. Checking for updates should now work without error. Enjoy!

Perhaps the Zettlr installer could be updated to include the .bat file (in the Zettlr install directory) and set the shortcut accordingly.

**Comment by hype8912** [comment]
Depending on the version of Electron being used this issue may be fixed in newer versions of Electron. 

https://github.com/electron/electron/issues/41590
https://github.com/electron/electron/pull/41689/commits

**Comment by brad** [comment]
Yes, this option is available with the electron version zettlr is currently using, but unfortunately you have to flip the [`nodeOptions` fuse](https://www.electronjs.org/docs/latest/tutorial/fuses#nodeoptions) to use it.

**Comment by nathanlesage** [comment]
Mh, that's a bummer -- because I don't want to enable fuses if it's not necessary. Those things are there for a reason ...

Also, if the only affected thing is the update check I don't think that would justify opening a hole in the security of the app.

**Comment by brad** [comment]
Individuals can turn it on with `sudo electron-fuses write --app /usr/bin/zettlr EnableNodeOptionsEnvironmentVariable=on`

**Comment by nathanlesage** [comment]
> Individuals can turn it on with `sudo electron-fuses write --app /usr/bin/zettlr EnableNodeOptionsEnvironmentVariable=on`

… although that would then invalidate the code signing. But I guess it's a workaround. In general, I would expect that, for setups in corporate environments with MITM proxies, updating Zettlr should be done centrally so that users can disable the update check altogether.

Sounds as if this issue is now fixed!
